### PR TITLE
[FLINK-13438][hive] Support date type in Hive

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

Fix our support for date type in Hive.

## Brief change log

  - Add shim method to convert date instances between Flink and Hive
  - Add tests to verify read/write date type

## Verifying this change

New test cases

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
cc @JingsongLi @KurtYoung @bowenli86 @xuefuz @zjuwangg 
It's quite similar to our support for timestamp. Please take a look, thanks.

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV100.java [hunk 1]
```diff
@@ -364,9 +366,29 @@ public LocalDateTime toFlinkTimestamp(Object hiveTimestamp) {
...
+	@Override
+	public Object toHiveDate(Object flinkDate) {
+		ensureSupportedFlinkDate(flinkDate);
+		return flinkDate instanceof Date ? flinkDate : Date.valueOf((LocalDate) flinkDate);
+	}
+
+	@Override
+	public LocalDate toFlinkDate(Object hiveDate) {
+		Preconditions.checkArgument(hiveDate instanceof Date,
+				"Expecting Hive timestamp to be an instance of %s, but actually got %s",
```
Comment (xuefuz) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV100.java]: c&p typo: timestamp => Date

Comment (lirui-apache) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV100.java]: nice catch :)

Comment (KurtYoung) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV100.java]: I will check in a hotfix to fix this typo

**Pull request closed without merging**